penn/studyspaces.py

```diff
@@ -0,0 +1,98 @@
+from bs4 import BeautifulSoup
+import requests
+
+
+BASE_URL = "http://libcal.library.upenn.edu"
+
+
+class StudySpaces(object):
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def date_parse(original):
+        """Parses the date to dashed format.
+
+        :param original: string with date in the format MM/DD/YYYY.
+        """
+        l = original.split("-")
+        final = [l[1], l[2], l[0]]
+        return '-'.join(final)
+
+    def get_id_json(self):
+        """Makes JSON with each element associating URL, ID, and building
+        name.
+        """
+        group_study_codes = []
+        url = BASE_URL + "/booking/vpdlc"
+        soup = BeautifulSoup(requests.get(url).text, 'lxml')
+        l = soup.find_all('option')
+        for element in l:
+            if element['value'] != '0':
+                url2 = BASE_URL + str(element['value'])
+                soup2 = BeautifulSoup(requests.get(url2).text, 'lxml')
```

Comment:
Why are you scraping with the 'lxml' library? This seems to be tripping up the tests, I haven't been able to test it locally yet. Any reason not to use the 'html5' library that's used for laundry?